### PR TITLE
Adds a small fishing pool on metastation

### PR DESCRIPTION
**Pull request by Maia-J**
## About The Pull Request

Firstly adds `var/datum/fish_source/fishing_datum` to `/turf/open/water`, allowing a much easier way of adding lootpools to water tiles, and sets its default to `/datum/fish_source/portal`

Secondly, adds a small pool for fishing in the metastation garden. This sadly means Betsy had to move in with the chickens.

## Why It's Good For The Game

Should show more people that fishing actually exists, and give you something more to do in the face of chaos and murder.

image

This is mostly a test run. I plan on following up with more pools on other stations if people like this one

## Changelog
:cl: @MrEmre12, @Majkl-J 
add: Fishing pool introduced to metastation garden
refactor: All water tiles now handle fishing through a unified fishing_datum variable
/:cl:

### CODE REVIEW

**Comment by Majkl-J** [comment]
> can't believe assistants are eating the decorative fish in the station's garden like stray cats.

The felinid attraction pool has been a disaster for tgstation

**Comment by DaCoolBoss** [comment]
It doesn't really sell itself as a functional fishing spot, maybe toss in a fishing toolbox to indicate what the pool is for?

**Comment by Majkl-J** [comment]
> It doesn't really sell itself as a functional fishing spot, maybe toss in a fishing toolbox to indicate what the pool is for?

Done with https://github.com/tgstation/tgstation/pull/82721/commits/2c8dcfe7e7fb07d49c9e229aaeb9d5cc3e6528bc

**Comment by EricZilla** [comment]
Would you consider making the pond 2x2, moving the disposal down by one, and making the grass area 2x4? That way the animals dont look so cramped

**Comment by OneAsianTortoise** [comment]
> Would you consider making the pond 2x2, moving the disposal down by one, and making the grass area 2x4? That way the animals dont look so cramped

Screw animal right ! stanchion barn all the way!

**Comment by Majkl-J** [comment]
> Would you consider making the pond 2x2, moving the disposal down by one, and making the grass area 2x4? That way the animals dont look so cramped

Done, see updated PR body. Both the disposals and APC seem to have worked fine

**Comment by EricZilla** [comment]
Very classy. Big fan.

**Comment by carlarctg** [comment]
Fishing is a wholesome group activity, this pool is relegated to a corner and just looks cramped and Bad. Feels weird

**Comment by MMMiracles** [comment]
i'm not against the idea of a fishing spot in a public area but as it stands it really is just a 2x2 water spot crammed into the corner. would like to see it more properly integrated with the commons area before a merge (not entirely sure how you'll do that without a decent rework of the entire room itself honestly, due to the space you're competing with).

**Comment by Majkl-J** [comment]
It's mostly limited by space restrictions, so sadly this is the best I can afford without shifting anything majorly around. Though, if this one seems to work and be fairly popular I'll work in one for most if not all maps, in a more proper permanent matter.

**Comment by jlsnow301** [comment]
dont like that its small. i think it should take over the spot in the rec room. not a maptainer to make that call, so just my opinion

**Comment by tralezab** [comment]
Can we at least pretend we're on a space station and put it in a way that kind of looks believable and justified instead of... I dunno, a minecraft infinite water source? Fantastic idea if the mapping is improved

**Review by MrMelbert — CHANGES_REQUESTED** [review]
100% needs decal work at the very least to make it blend in with the setting.

**Comment by Majkl-J** [comment]
Updated with new looks yesterday, see PR body

**Comment by MrEmre12** [comment]
I loved your idea! If you like, you can create the pool like this, it's really great.

Ekran görüntüsü 2024-05-02 195248

**Comment by MrMelbert** [comment]
> I loved your idea! If you like, you can create the pool like this, it's really great.

This is a good way of integrating it, I would prefer if we ran with this

**Comment by Majkl-J** [comment]
> > I loved your idea! If you like, you can create the pool like this, it's really great.
> 
> This is a good way of integrating it, I would prefer if we ran with this

Likewise, that's much better work than mine

**Comment by Majkl-J** [comment]
> I loved your idea! If you like, you can create the pool like this, it's really great.

Care to direct the map file my way? It's much better work than I could've ever pulled off, discord's @thewaterpig 

(Obviously Ill write you into the changelog)

**Comment by Majkl-J** [comment]
Updated
image

File: code/game/turfs/open/water.dm [hunk 1]
```diff
@@ -37,6 +42,4 @@
 	baseturfs = /turf/open/water/beach
 	immerse_overlay_color = "#7799AA"
```
Comment (Maurukas) [inline code/game/turfs/open/water.dm:44]:
```suggestion
```
extra line here

**Review by Maurukas — APPROVED** [review]

**Pull request merged**